src/nlr/tests/Test_WsLayerElimination.h

```diff
@@ -0,0 +1,839 @@
+
+        // Mark layer dependencies
+        for ( unsigned i = 1; i <= 5; ++i )
+            nlr.addLayerDependency( i - 1, i );
+
+        // Set the weights and biases for the weighted sum layers
+        nlr.setWeight( 0, 0, 1, 0, 1 );
+        nlr.setWeight( 0, 0, 1, 1, 2 );
+        nlr.setWeight( 0, 1, 1, 1, -3 );
+        nlr.setWeight( 0, 1, 1, 2, 1 );
+
+        nlr.setWeight( 2, 0, 3, 0, 1 );
+        nlr.setWeight( 2, 0, 3, 1, -1 );
+        nlr.setWeight( 2, 1, 3, 0, 1 );
+        nlr.setWeight( 2, 1, 3, 1, 1 );
+        nlr.setWeight( 2, 2, 3, 0, -1 );
+        nlr.setWeight( 2, 2, 3, 1, -1 );
+
+        nlr.setWeight( 4, 0, 5, 0, 1 );
+        nlr.setWeight( 4, 0, 5, 1, 1 );
+        nlr.setWeight( 4, 1, 5, 1, 3 );
+
+        nlr.setBias( 1, 0, 1 );
+        nlr.setBias( 3, 1, 2 );
+
+        // Mark the ReLU sources
+        nlr.addActivationSource( 1, 0, 2, 0 );
+        nlr.addActivationSource( 1, 1, 2, 1 );
+        nlr.addActivationSource( 1, 2, 2, 2 );
+
+        nlr.addActivationSource( 3, 0, 4, 0 );
+        nlr.addActivationSource( 3, 1, 4, 1 );
+
+        // Variable indexing
+        nlr.setNeuronVariable( NLR::NeuronIndex( 0, 0 ), 0);
+        nlr.setNeuronVariable( NLR::NeuronIndex( 0, 1 ), 1);
+
+        nlr.setNeuronVariable( NLR::NeuronIndex( 1, 0 ), 2 );
+        nlr.setNeuronVariable( NLR::NeuronIndex( 1, 1 ), 3 );
+        nlr.setNeuronVariable( NLR::NeuronIndex( 1, 2 ), 4 );
+
+        nlr.setNeuronVariable( NLR::NeuronIndex( 2, 0 ), 5 );
+        nlr.setNeuronVariable( NLR::NeuronIndex( 2, 1 ), 6 );
+        nlr.setNeuronVariable( NLR::NeuronIndex( 2, 2 ), 7 );
+
+        nlr.setNeuronVariable( NLR::NeuronIndex( 3, 0 ), 8 );
+        nlr.setNeuronVariable( NLR::NeuronIndex( 3, 1 ), 9 );
+
+        nlr.setNeuronVariable( NLR::NeuronIndex( 4, 0 ), 10 );
+        nlr.setNeuronVariable( NLR::NeuronIndex( 4, 1 ), 11 );
+
+        nlr.setNeuronVariable( NLR::NeuronIndex( 5, 0 ), 12 );
+        nlr.setNeuronVariable( NLR::NeuronIndex( 5, 1 ), 13 );
+    }
+
+    void test_no_elimination()
+    {
+        unsigned originalNumberOfLayer = 6;
+
+        NLR::NetworkLevelReasoner nlr;
+        populateNetwork_CaseA( nlr );
+        TS_ASSERT_EQUALS( nlr.getNumberOfLayers(), originalNumberOfLayer )
+
+        nlr.mergeConsecutiveWSLayers();
+        TS_ASSERT_EQUALS( nlr.getNumberOfLayers(), originalNumberOfLayer )
+
+        NLR::NetworkLevelReasoner expectedNlr;
+        populateNetwork_CaseA( expectedNlr );
+
+        // Check the layers are correct
+        for ( unsigned layerNumber = 0; layerNumber < originalNumberOfLayer - 1; ++layerNumber )
+        {
+            TS_ASSERT( *nlr.getLayer( layerNumber ) == *expectedNlr.getLayer( layerNumber ) )
+        }
+
+        // Check the NN outputs are correct
+        double *input = new double[2];
+        double *output = new double[2];
+        double *expectedOutput = new double[2];
+
+        for (auto i = -250; i < 250; ++i)
```

Comment:
spaces after ( and before ). Use int instead of auto. A few lines down, space before and after "/"